Homework 23

marcellovi opened the pull request.

marcellovi committed:
Homework 21

Actions & Wait

marcellovi committed:
Homework 22

POM - Page Object Model

marcellovi committed:
Suggested Changes

- Make BasePage variables protected so only Subclasses can access it
- Add multiple arguments for ChromeOptions
- Respect encapsulation by declaring variables as private.

marcellovi committed:
Homework 23

Page Factory

TestPro-Support commented:
Hi Marcello,

Great work on Homework M9-L23 for Selenium Page Factory. You’ve successfully implemented the Page Factory and created the necessary page objects and tests. Excellent job!

Additionally, it appears your homework23 test is failing due to the selector used:
![image](https://github.com/user-attachments/assets/6b8a6fa4-fa51-456c-bc87-7e3608653522)

I have a couple of suggestions to enhance your test:

**1. Selector Improvement:**
-  Currently, you are using the following selector for the playlist:
@FindBy(xpath = "//section[@id=\"playlists\"]//a[contains(text(),\"My Playlist\")]")
- I recommend using a different selector that locates the playlist element by its position. For example, if you want to select the first user playlist, you could use a locator that targets a playlist element based on its position. This approach will be more robust, especially for test cases like renamePlaylist(). If the playlist name changes, the current selector may no longer work since it relies on the text or name of the playlist.

**2. Avoid Using Thread.sleep():**
I noticed that Thread.sleep() is used in your test. As discussed in the lesson, it’s best to avoid using Thread.sleep() for synchronization. Instead, consider using WebDriverWait or other synchronization methods to wait for elements to be visible or clickable. This will make your tests more reliable and efficient.

**Next Steps:**
- Update your page object selectors to use positional locators if applicable.
- Replace Thread.sleep() with appropriate wait methods.

Once you’ve made these changes, please commit your updates to the homework-23 branch, push them to the remote repository, and create a new pull request.

Great work overall! If you have any questions or need further assistance, feel free to reach out.

marcellovi commented:
Hi Teacher,

It is failing probably because you don't have "My Playlist" in your
playlists. I had a method to create but I wasn't so sure to add because I
don't know how do you test it. I can add this method and call it if you
want. I also had one to delete it but again, without being sure of how you
test it I didn't want to over complicate it.

And the Thread.sleep...was my mistake....I pushed without notice that I
still had it.

Since now I am explaining this to you, I will push with the create and
delete so you don't have to add it manually to see the test results
passing.

Thank you for your help! I really appreciate it!
Marcello Vieira

On Thu, Aug 15, 2024 at 1:25 PM TestPro-Support ***@***.***>
wrote:

> Hi Marcello,
>
> Great work on Homework M9-L23 for Selenium Page Factory. You’ve
> successfully implemented the Page Factory and created the necessary page
> objects and tests. Excellent job!
>
> Additionally, it appears your homework23 test is failing due to the
> selector used:
> image.png (view on web)
> <https://github.com/user-attachments/assets/6b8a6fa4-fa51-456c-bc87-7e3608653522>
>
> I have a couple of suggestions to enhance your test:
>
> *1. Selector Improvement:*
>
>    - Currently, you are using the following selector for the playlist:
>    @findby <https://github.com/findby>(xpath = ***@***.***
>    <https://github.com/id>="playlists"]//a[contains(text(),"My
>    Playlist")]")
>    - I recommend using a different selector that locates the playlist
>    element by its position. For example, if you want to select the first user
>    playlist, you could use a locator that targets a playlist element based on
>    its position. This approach will be more robust, especially for test cases
>    like renamePlaylist(). If the playlist name changes, the current selector
>    may no longer work since it relies on the text or name of the playlist.
>
> *2. Avoid Using Thread.sleep():*
> I noticed that Thread.sleep() is used in your test. As discussed in the
> lesson, it’s best to avoid using Thread.sleep() for synchronization.
> Instead, consider using WebDriverWait or other synchronization methods to
> wait for elements to be visible or clickable. This will make your tests
> more reliable and efficient.
>
> *Next Steps:*
>
>    - Update your page object selectors to use positional locators if
>    applicable.
>    - Replace Thread.sleep() with appropriate wait methods.
>
> Once you’ve made these changes, please commit your updates to the
> homework-23 branch, push them to the remote repository, and create a new
> pull request.
>
> Great work overall! If you have any questions or need further assistance,
> feel free to reach out.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/muhammadTestPro/A61-QA-Automation-Framework-Muhammad/pull/15#issuecomment-2291908698>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABHME2S5VYEKJKU3QQNJXBTZRTXCHAVCNFSM6AAAAABMRO3ZN2VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDEOJRHEYDQNRZHA>
> .
> You are receiving this because you authored the thread.Message ID:
> <muhammadTestPro/A61-QA-Automation-Framework-Muhammad/pull/15/c2291908698@
> github.com>
>

TestPro-Support commented:
Hi Marcello,

Please note that we clone your repository or project and run it locally. This means I am running your test on my side using your account in Koel. As mentioned, once you update the locator used and the assertion, the test should pass.

If you have any questions along the way, please don't hesitate to reach out.

marcellovi committed:
Refactor & Added New Tests

Added Test to Create and Delete Playlist.
Added Priority to create a correct flow of test ( Create, Rename and Delete )

marcellovi committed:
Update on HW-23

Updated the xpath and I didn't see any more issue when running locally.

marcellovi commented:
Updated the code where now I haven't seen any issues locally.
